### PR TITLE
FSET-1710. Select reason for candidate removal from event

### PR DESCRIPTION
**Pull request by vglushak**

### CODE REVIEW

File: app/repositories/CandidateAllocationRepository.scala [hunk 1]
```diff
@@ -54,17 +58,49 @@ class CandidateAllocationMongoRepository(implicit mongo: () => DB)
...
+      val jsObjects = allocations.map(format.writes)
+      collection.bulkInsert(jsObjects.toStream, ordered = false)
+    } map ( _ => () )
   }
 
-  def allocationsForEvent(eventId: String): Future[Seq[CandidateAllocation]] = {
-    collection.find(BSONDocument("eventId" -> eventId), projection)
+  def findCandidateRemovals(applications: Seq[String]): Future[Seq[CandidateAllocation]] = {
+    collection.find(BSONDocument(
+      // "id" -> BSONDocument("$in" -> applications),
```
Comment (kjharkin) [inline app/repositories/CandidateAllocationRepository.scala]: remove comment

File: app/services/allocation/CandidateAllocationService.scala [hunk 2]
```diff
@@ -130,6 +132,39 @@ trait CandidateAllocationService extends EventSink {
     }
   }
 
+  def findCandidatesEligibleForEventAllocation(assessmentCenterLocation: String) = {
+    applicationRepo.findCandidatesEligibleForEventAllocation(List(assessmentCenterLocation)).flatMap { resp =>
+      candidateAllocationRepo.findNoShowAllocations(resp.candidates.map(_.applicationId)).map { noShowCandidates =>
+        val noShowCandidatesIds = noShowCandidates.map(_.id)
+        val resCandidates = resp.candidates.filter(c => !noShowCandidatesIds.contains(c.applicationId))
+        val c = resp.copy(candidates = resCandidates, totalCandidates = resCandidates.size)
+        c
```
Comment (kjharkin) [inline app/services/allocation/CandidateAllocationService.scala]: tidy up - don't need ```c```

File: conf/app.routes [hunk 3]
```diff
@@ -137,14 +135,20 @@ PUT         /assessor/:assessorId/allocations                            control
...
-GET         /events/:eventId/sessions/:sessionId/candidateAllocations    controllers.EventsController.getCandidateAllocations(eventId: String, sessionId: String)
-POST        /events/:eventId/sessions/:sessionId/remove-candidates       controllers.EventsController.removeCandidateAllocations(eventId: String, sessionId: String)
+
+# candidate allocations
+GET         /candidate-allocations/find-applications/location/:location            controllers.CandidateAllocationController.findCandidatesEligibleForEventAllocation(location: String)
+POST        /candidate-allocations/find-allocated                                  controllers.CandidateAllocationController.findAllocatedApplications()
+GET         /candidate-allocations/events/:eventId/sessions/:sessionId             controllers.CandidateAllocationController.getCandidateAllocations(eventId: String, sessionId: String)
+POST        /candidate-allocations/remove/events/:eventId/sessions/:sessionId      controllers.CandidateAllocationController.removeCandidateAllocations(eventId: String, sessionId: String)
```
Comment (kjharkin) [inline conf/app.routes]: why is this not a DELETE?

**Comment by kjharkin** [comment]
Couple of relatively minor comments - will merge once addressed. Good stuff.

File: app/repositories/CandidateAllocationRepository.scala (hunk 1 above)
Comment (vglushak) [inline app/repositories/CandidateAllocationRepository.scala]: Done.

File: app/services/allocation/CandidateAllocationService.scala (hunk 2 above)
Comment (vglushak) [inline app/services/allocation/CandidateAllocationService.scala]: Done.

File: conf/app.routes (hunk 3 above)
Comment (vglushak) [inline conf/app.routes]: List of applicationIds are passed as body into this request.  Body is not supported in DELETE request.

**Comment by vglushak** [comment]
Comments addressed where possible.

**Pull request merged**